packages/next/client/image.tsx

```diff
@@ -657,6 +662,26 @@ export default function Image({
     [imageSizesPropName]: imgAttributes.sizes,
   }
 
+  const useLayoutEffect =
+    typeof window === 'undefined' ? React.useEffect : React.useLayoutEffect
+  const onLoadingCompleteRef = useRef(onLoadingComplete)
+
+  useEffect(() => {
+    onLoadingCompleteRef.current = onLoadingComplete
+  }, [onLoadingComplete])
+
+  useLayoutEffect(() => {
+    if (ref.current) {
+      setIntersection(ref.current)
+    }
+  }, [setIntersection])
+
+  useEffect(() => {
+    if (ref.current) {
+      handleLoading(ref, srcString, layout, placeholder, onLoadingCompleteRef)
+    }
+  }, [srcString, layout, placeholder, isVisible])
```

Comment:
Will `handleLoading` be triggered again when any of these props changes e.g. `layout`?

Reply:
I added a test for that in 8c2f1eb 👍